fastify.js

```diff
@@ -585,6 +590,16 @@ function build (options) {
   function addHook (name, fn) {
     throwIfAlreadyStarted('Cannot call "addHook" when fastify instance is already started!')
 
+    if (name === 'onSend' || name === 'preSerialization') {
+      if (fn.constructor.name === 'AsyncFunction' && fn.length === 4) {
+        throw new FST_ERR_HOOK_INVALID_FUNCTION()
+      }
+    } else {
+      if (fn.constructor.name === 'AsyncFunction' && fn.length === 3) {
+        throw new FST_ERR_HOOK_INVALID_FUNCTION()
```

Comment:
Throwing an error here would be a breaking change. Maybe log a warning for now and then throw in the next major?

Reply:
Good point :)